File: artifacts/api-server/src/routes/github.ts
import { Router, type IRouter } from "express";
import {
  GetGithubProfileResponse,
  ListGithubReposResponse,
} from "@workspace/api-zod";

const router: IRouter = Router();

// Static GitHub data — in production you'd cache calls to the GitHub API
const GITHUB_PROFILE = {
  username: "aniketpaswan",
  bio: "Backend Engineer building scalable systems. B.Tech CSE | FastAPI | PostgreSQL | Docker",
  publicRepos: 18,
  followers: 42,
  following: 31,
  avatarUrl: "https://avatars.githubusercontent.com/u/aniketpaswan",
  htmlUrl: "https://github.com/aniketpaswan",
  totalStars: 87,
  totalCommits: 412,
  topLanguages: ["Python", "TypeScript", "SQL", "Bash"],
};

const GITHUB_REPOS = [
  {
    name: "api-security-platform",
    description: "A FastAPI middleware that protects APIs using behavioral analysis and adaptive risk scoring",
    url: "https://github.com/aniketpaswan/api-security-platform",
    stars: 34,
    forks: 7,
    language: "Python",
    topics: ["fastapi", "security", "redis", "postgresql"],
  },
  {
    name: "auth-system",
    description: "High-performance authentication system with JWT, Redis caching and role-based access control",
    url: "https://github.com/aniketpaswan/auth-system",
    stars: 28,
    forks: 5,
    language: "Python",
    topics: ["jwt", "redis", "fastapi", "postgresql"],
  },
  {
    name: "mental-health-chatbot",
    description: "An AI-powered mental health support chatbot fine-tuned using T5 model and built with modern web stack",
    url: "https://github.com/aniketpaswan/mental-health-chatbot",
    stars: 25,
    forks: 4,
    language: "TypeScript",
    topics: ["ai", "mongodb", "react", "tailwindcss"],
  },
];

router.get("/github/profile", async (_req, res): Promise<void> => {
  res.json(GetGithubProfileResponse.parse(GITHUB_PROFILE));
});

router.get("/github/repos", async (_req, res): Promise<void> => {
  res.json(ListGithubReposResponse.parse(GITHUB_REPOS));
});

export default router;

File: backend/routers/github.py
from fastapi import APIRouter
from schemas import GithubProfile, GithubRepo

router = APIRouter()

_PROFILE = GithubProfile(
    username="anikk0208",
    bio="Backend Engineer | B.Tech CSE @ AKTU | FastAPI • PostgreSQL • Redis • Docker",
    public_repos=12,
    followers=18,
    following=24,
    avatar_url="https://avatars.githubusercontent.com/u/anikk0208",
    html_url="https://github.com/anikk0208",
    total_stars=62,
    total_commits=380,
    top_languages=["Python", "SQL", "JavaScript", "Bash"],
)

_REPOS = [
    GithubRepo(
        name="api-security-middleware",
        description="Behavior-based API security middleware with 10+ independent components — policy evaluation, trust scoring, adaptive enforcement. 16–20 ms average latency using Redis.",
        url="https://github.com/anikk0208/api-security-middleware",
        stars=28,
        forks=5,
        language="Python",
        topics=["fastapi", "security", "redis", "postgresql", "middleware"],
    ),
    GithubRepo(
        name="fastapi-auth-system",
        description="Modular authentication system with JWT, Redis caching, role-based authorization, and email verification. Handles ~800 concurrent users.",
        url="https://github.com/anikk0208/fastapi-auth-system",
        stars=21,
        forks=4,
        language="Python",
        topics=["fastapi", "jwt", "redis", "postgresql", "authentication"],
    ),
    GithubRepo(
        name="mental-health-chatbot",
        description="AI-powered mental health chatbot using fine-tuned T5 Transformer with semantic retrieval, 99.2% safety rate.",
        url="https://github.com/anikk0208/mental-health-chatbot",
        stars=13,
        forks=2,
        language="Python",
        topics=["pytorch", "nlp", "mongodb", "huggingface", "transformers"],
    ),
]


@router.get("/github/profile", response_model=GithubProfile)
def get_github_profile():
    return _PROFILE


@router.get("/github/repos", response_model=list[GithubRepo])
def list_github_repos():
    return _REPOS
